src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java

```diff
@@ -163,6 +168,72 @@ protected boolean submit(O op) throws InterruptRunException, IllegalStateExcepti
         return false;
     }
 
+    protected boolean submitItemCreate(final DataOperation<? extends DataItem> eventOperation, final DataItem eventItem)
+    {
+        final String path = eventOperation.dstPath();
+        final String scope = path.substring(0, path.indexOf("/"));
+        final String streamName = path.substring(path.indexOf("/")+1);
+
+        if (scopeMap.get(scope).get(streamName) == null) {
+            Loggers.ERR.debug("Failed to create event: the stream {}  not found in the scope {}", streamName, scope);
+            eventOperation.status(Operation.Status.RESP_FAIL_UNKNOWN);
+            return false;
+        };
+
+        StreamConfiguration streamConfig = StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.fixed(1))
+                .build();
+
+        streamManager.createStream(scope, streamName, streamConfig);
+
+        try (ClientFactory clientFactory = ClientFactory.withScope(scope, URI.create(uriSchema));
+             EventStreamWriter<ByteBuffer> writer = clientFactory.createEventWriter(streamName,
+                     null,
+                     EventWriterConfig.builder().build())) {
+            final String routingKey = null;
+
+
+            final long buffSize;
+            try {
+                buffSize = eventItem.size();
+            } catch (IOException e) {
+                eventOperation.status(Operation.Status.FAIL_IO);
+                Loggers.ERR.debug("Failed to create event: " + e.getMessage());
+                return false;
+            }
+
+            final DataInput dataInput = eventItem.dataInput();
+            final long offset = eventItem.offset();
+            final int dataSize = dataInput.getSize();
+
+            final ByteBuffer outBuff = dataInput
+                    .getLayer(eventItem.layer())
+                    .asReadOnlyBuffer();
+
+            outBuff.position((int)offset);
+            outBuff.limit(dataSize);
```

Comment:
Will throw an `IllegalArgumentException` if the `dataSize` value is more than the remaining byte count in the buffer. The limit value should be a minimum of two:
- dataSize
- buffer remaining bytes
Let's save that minimum into the new `int n` local variable and use it instead of the `dataSize` further in this method.